src/shared/settingsConfiguration.ts

```diff
@@ -31,10 +47,15 @@ export class DefaultSettingsConfiguration implements SettingsConfiguration {
         return val ?? defaultValue
     }
 
-    public async writeSetting<T>(settingKey: string, value: T, target: vscode.ConfigurationTarget): Promise<void> {
+    public async writeSetting<T>(settingKey: string, value: T, target: vscode.ConfigurationTarget): Promise<boolean> {
         // tslint:disable-next-line:no-null-keyword
-        const settings = vscode.workspace.getConfiguration(this.extensionSettingsPrefix, null)
-
-        await settings.update(settingKey, value, target)
+        try {
+            const settings = vscode.workspace.getConfiguration(this.extensionSettingsPrefix)
+            await settings.update(settingKey, value, target)
+            return true
+        } catch (e) {
+            getLogger().error('failed to set config: %O=%O, error: %O', settingKey, value, e)
+            return false
```

Comment:
this prevents "Toolkit failed to activate" error

Reply:
Are there any `writeSetting` calls that should take action when false is returned? 

For example, if certain settings are not applied, will the toolkit stability be compromised and should users be informed about this?

Reply:
That is something I planned for a separate followup, since it will involve potential UX considerations and may involves refactoring (to bubble up the result to higher layers)

Reply:
Ok. Please add a todo into the code and ensure there is a backlog entry.